share/android/src/main/java/com/capacitorjs/plugins/share/SharePlugin.java

```diff
@@ -37,6 +37,17 @@ public void onReceive(Context context, Intent intent) {
         }
     }
 
+    @ActivityCallback
+    private void activityResult(PluginCall call, ActivityResult result) {
+        if (result.getResultCode() == Activity.RESULT_CANCELED) {
```

Comment:
```suggestion
        if (result.getResultCode() == Activity.RESULT_CANCELED && !stopped) {
```
It's missing the ` && !stopped` check. Without it, when you open an app to complete the share and get back to the app the promise is rejected instead of resolving with the app package name.